packages/js/product-editor/src/hooks/use-product.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * External dependencies
+ */
+import { applyFilters } from '@wordpress/hooks';
+import { Product, ReadOnlyProperties } from '@woocommerce/data';
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore No types for this exist yet.
+// eslint-disable-next-line @woocommerce/dependency-group
+import { useEntityId } from '@wordpress/core-data';
+import { useDispatch } from '@wordpress/data';
+
+
+export function useProduct() {
```

Comment:
Adding an optional `id` argument may be helpful, but we can do this in a follow-up.